### PR TITLE
fix: serve handler error with 0 arguments

### PR DESCRIPTION
**Pull request by marvinhagemeister**
Fixes https://github.com/denoland/deno/issues/23651

### CODE REVIEW

**Review by mmastrac — APPROVED** [review]
LGTM. I'm OK to merge this, but it's unlikely that a user would realistically write a handler with no arguments and this helps avoid cases where a `fetch` export sets off a false positive for our `deno serve` hint message. If we're good with that tradeoff, we can merge it. :)

File: tests/specs/serve/no_args/__test__.jsonc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "args": "serve --port 12345 main.ts",
```
Comment (dsherret) [inline tests/specs/serve/no_args/__test__.jsonc:2]: Does `deno serve` find an open port? Maybe we should exclude it from the test? This port conflicts with another test.

Comment (satyarohith) [reply tests/specs/serve/no_args/__test__.jsonc:2]: Is there a way to pick to an open port here? `deno serve` doesn't find an open port.

Comment (satyarohith) [reply tests/specs/serve/no_args/__test__.jsonc:2]: This should fix it https://github.com/denoland/deno/pull/23846

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**